planemo/commands/cmd_shed_upload.py

```diff
@@ -102,8 +103,33 @@ def __handle_upload(ctx, path, **kwds):
     message = kwds.get("message", None)
     if message:
         update_kwds["commit_message"] = message
+
+    repo_id = __find_repository(ctx, tsi, path, kwds)
+    if repo_id is None:
+        return -1
+
+    try:
+        tsi.repositories.update_repository(repo_id, tar_path, **update_kwds)
+    except Exception as e:
+        exception_content = e.read()
+        try:
+            upstream_error = json.loads(exception_content)
+            error(upstream_error['err_msg'])
+        except:
```

Comment:
I've switched to reading the exception, hope that meets your requirements. Not a huge improvement, but it (was) done elsewhere in that file previously, so it seemed like it'd be ok? I haven't had time to debug exactly what was causing the failure to read the error message in the second bare exception that was patched, for future debugging, it happened in `package_vienna_*` in the tools_iuc repo.